### ISSUE TITLE
Customizer: Out of place tab if double allocation of a variable of this group

### ISSUE THREAD

**Issue by PRouzeau** [openscad/openscad#2439]
label [Customizer]
Bug:
If a variable is allocated a first time, then the same variable is allocated a second time within a customizer group, the group tab is moved to the position of the first allocation, whatever its original position is supposed to be. In principle, it is not normal to allocate a variable twice when using customizer, but it can be difficult to track it without a message. This bug may not need to be removed, provided a message about the double allocation is sent.

**Comment by MichaelPFrey** [comment]
Code showing the issue:
```
a=5;
/* [Group 1] */
a=10;
b=10;
/* [Group 2] */
a=20;
```
screenshot:
![bildschirmfoto von 2018-08-25 08-09-02](https://user-images.githubusercontent.com/24962768/44615603-3971d400-a83e-11e8-8e68-94f4ea988864.png)
hmmm...

**Comment by t-paul** [comment]
Reassingment should ideally be an error. Right now it's not even a warning. It's not as easy as it sounds as there are some usecases that historically mis-use that feature (e.g. the -D command line parameter).

This also seems to be a duplicate of #1770.

**Comment by MichaelPFrey** [comment]
every assignment knows its location:
`const Location loc =assignment.location();`
which includes `firstLine()`. `ParameterExtractor::setParameters` gets the assignments in order of first assignment, but with the information of the last assignment. We could reorder the assignments by location. That would at least fix the "out of place" placement of the group. std::sort with a custom function could be used for that.

@t-paul good catch - I can see it in two ways: Fixing #1770 also solves #2439. But #2439 could be solved/worked around without resolving #1770, as I see some issues with the customizer code as well.
Well, I start working on #1770. I see away to deal with include and the -D command line parameter.

**Comment by t-paul** [comment]
Right, good point. Let's leave this open then as separate topic to sync data and location of the assignments.

**Comment by MichaelPFrey** [comment]
Testing my patch, I found `value-reassignment-tests.scad`. This test case has a similar root cause to this issue. But thinking about it, it shows how complicated rearranging the statements would get.
